opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -139,7 +140,7 @@ def __init__(
         attributes: types.Attributes = None,
     ) -> None:
         super().__init__(context)
-        self._attributes = attributes
+        self._attributes = _create_immutable_attributes(attributes)
```

Comment:
Can you update the docstring for Link and Event about immutability like we do for Resource?

Reply:
Good call, done.